Source/WebCore/css/CSSBackgroundRepeatValue.cpp

```diff
@@ -44,17 +44,27 @@ Ref<CSSBackgroundRepeatValue> CSSBackgroundRepeatValue::create(CSSValueID repeat
     return adoptRef(*new CSSBackgroundRepeatValue(repeatXValue, repeatYValue));
 }
 
-String CSSBackgroundRepeatValue::customCSSText() const
+template<typename Maker> decltype(auto) CSSBackgroundRepeatValue::serialize(Maker&& maker) const
 {
     // background-repeat/mask-repeat behave a little like a shorthand, but `repeat no-repeat` is transformed to `repeat-x`.
     if (m_xValue != m_yValue) {
         if (m_xValue == CSSValueRepeat && m_yValue == CSSValueNoRepeat)
-            return nameString(CSSValueRepeatX);
+            return maker(nameString(CSSValueRepeatX));
```

Comment:
There’s a possible efficiency issue here. If we want a string, then `nameString` is perfect since the string is already allocated. If we just want to append then `nameLiteral` is slightly more efficient.